server/api-util/lineItems.js

```diff
@@ -140,18 +155,23 @@ exports.transactionLineItems = (listing, orderData, providerCommission, customer
   const quantityAndExtraLineItems =
     unitType === 'item'
       ? getItemQuantityAndLineItems(orderData, publicData, currency)
-      : unitType === 'hour'
+      : unitType === 'hour' && !orderData.seats
       ? getHourQuantityAndLineItems(orderData)
-      : ['day', 'night'].includes(unitType)
+      : unitType === 'hour'
+      ? getHoursWithSeatsAndLineItems(orderData)
+      : ['day', 'night'].includes(unitType) && !orderData.seats
       ? getDateRangeQuantityAndLineItems(orderData, code)
+      : ['day', 'night'].includes(unitType)
+      ? getDateRangeWithSeatsAndLineItems(orderData, code)
       : {};
 
-  const { quantity, extraLineItems } = quantityAndExtraLineItems;
+  const { quantity, units, seats, extraLineItems } = quantityAndExtraLineItems;
 
+  // TODO: We could use a better error message here.
   // Throw error if there is no quantity information given
-  if (!quantity) {
+  if (!quantity && !(units && seats)) {
     const message = `Error: transition should contain quantity information: 
-      stockReservationQuantity, quantity, or bookingStart & bookingEnd (if "line-item/day" or "line-item/night" is used)`;
+       stockReservationQuantity, quantity, units & seats, or bookingStart & bookingEnd (if "line-item/day" or "line-item/night" is used)`;
```

Comment:
This line doesn't make sense. At least, the 'if' is off.
Perhaps it could be removed or changed somehow.